### PR TITLE
stop implictly adding repos using distro matches (#327)

### PR DESCRIPTION
**Pull request by rkent**
As discussed in #327, we want to move away from aggressively trying to locate possible releases of ros packages, based on finding a branch name that contains a distro name, and instead rely on rosdistro for repos.

One minor drive-by change: while investigating some earlier problems, I found release repos that had changed their branch name to main, so I added that to the release repo search.

I've run the index using these changes, and uploaded (temporarily) to http://test1.rosdabbler.com

### CODE REVIEW

File: _plugins/rosindex_generator.rb [hunk 1]
```diff
@@ -1082,7 +1082,7 @@ def generate(site)
                       end
 
                       # get the tracks file
-                      ['master','bloom'].each do |branch_name|
+                      ['master','bloom', 'main'].each do |branch_name|
```
Comment (tfoote) [inline _plugins/rosindex_generator.rb]: Is this change related?

File: _ruby_libs/rosindex.rb [hunk 2]
```diff
@@ -79,7 +79,7 @@ def initialize(name, type, uri, purpose, checkout_path, id = nil)
 
     # hash distro -> RepoSnapshot
     # each entry in this hash represents the preferred version for a given distro in this repo
-    @snapshots = Hash[$all_distros.collect { |d| [d, RepoSnapshot.new(nil, d, false, false)] }]
+    @snapshots = {}
```
Comment (tfoote) [inline _ruby_libs/rosindex.rb:82]: Is this the beginning of the orphaned code cleanup? Should we maybe defer that to later? I see that we're still using the snapshots in the main loop: https://github.com/ros-infrastructure/rosindex/pull/352/files#diff-70c045b5c7c6b1ab5c817acfead858a20af39af4e4724acc3c43caa0c0b20497R698

File: _plugins/rosindex_generator.rb [hunk 3]
```diff
@@ -711,7 +711,7 @@ def scrape_repo(site, repo)
         unless explicit_version.nil?
           dputs (" Looking for explicit version #{explicit_version}").green
         end
-        version, snapshot.version = vcs.get_version(distro, explicit_version)
+        version, snapshot.version = vcs.get_version(explicit_version)
```
Comment (tfoote) [inline _plugins/rosindex_generator.rb:714]: The second output from this command looks to be set to be the input in L701 above. And we're returning `nil, nil` or `branch.name, branch_name` where we've confirmed that `branch_name == explict_version` ala L297 below. 

Thus I think we can reduce this to just a single return of the version instead and test that for nil anywhere necessary as snapshot.version is always explict_version unless nil when version is also nil

File: _plugins/rosindex_generator.rb (hunk 1 above)
Comment (rkent) [reply _plugins/rosindex_generator.rb]: This is the minor drive-by change that I mentioned in the PR intro. While investigating the code to understand why some repos were or were not being included, I came across a ros gbp repo that used "main" instead of "master" as its branch name. Unfortunately I don't see any documentation in my notes about which one this was. I'll remove this change.

File: _ruby_libs/rosindex.rb (hunk 2 above)
Comment (rkent) [reply _ruby_libs/rosindex.rb:82]: This code adds a blank repo snapshot that matches all existing distro names, when a Repo instance is created. Later, when the release data is scanned, the snapshots for official distros are replaced with correct data. Then eventually all of these blank instances are aggressively searched for, trying to match a branch with the distro names. It is these searched-for instances that this PR is eliminating. AFAICT, there is no use for these blank instances other than to feed the aggressive search for the instances, that this PR is eliminating.

So yes this is orphaned code in the sense that later I could tell the difference between an officially released and a searched-for snapshot, and not use the searched-for instances. But I don't see the point in leaving blank instances here that are not used for anything (unless I am mistaken in their use).

File: _plugins/rosindex_generator.rb (hunk 3 above)
Comment (rkent) [reply _plugins/rosindex_generator.rb:714]: Yes I can see that the two returned values are now always equal, so this code can be further reduced. I'll do that.

Comment (rkent) [reply _plugins/rosindex_generator.rb:714]: Looking at this in more detail, I'm going to reverse what I said, and recommend against a change from my original patch here. snapshot.version is not always explicit version, for example in the Hg case, snapshot.version is 'default' when explicit_version is 'REMOTE_HEAD'. In the git version, it is more complex but not equal. Also, the two items returned are only both equal and strings in the Hg version. In the git version, the second item is a string, while the first is a Rugged branch object. It is possible that we could reduce all that to a single output, but that is really a refactor, not just eliminating redundant code.  Also, the snapshot.version that is returned from get_version may differ from the one earlier, as the name for example may be shortened by a split.

File: _plugins/rosindex_generator.rb (hunk 1 above)
Comment (tfoote) [reply _plugins/rosindex_generator.rb]: Ahh, thanks I hadn't looked deep enough in. I believe that bloom still assumes `master` as the name I think. This is something that could be extended here for future compatibility.

**Review by tfoote — APPROVED** [review]
Thanks for simplifying this here. Hopefully this will avoid unexpected results in the index.
